app/src/main/java/protect/budgetwatch/PasswordManager.java

```diff
@@ -0,0 +1,98 @@
+package protect.budgetwatch;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+
+public class PasswordManager {
+
+    private final String PasswordKey = "password";
+    private final int Iterations = 100;
+    private final int KeyLength = 128;
+
+    private SharedPreferences _prefs;
+
+    PasswordManager(Context context) {
+        _prefs = context.getSharedPreferences("protect.budgetwatch", Context.MODE_PRIVATE);
+    }
+
+    private String getStoredPasswordHash() {
+        return _prefs.getString(PasswordKey, null);
+    }
+
+    public void setPassword(String password) {
+        try {
+            // Generate Salt
+            SecureRandom sr = SecureRandom.getInstance("SHA1PRNG");
+            byte[] salt = new byte[KeyLength / 8];
+            sr.nextBytes(salt);
+            // Generate Hash
+            byte[] hash = generateHash(password, salt);
+            // Write to SharedPreferences
+            _prefs.edit().putString(PasswordKey, toHex(salt) + ":" + toHex(hash)).commit();
+        } catch (Exception e) {
```

Comment:
The issue is i don't know what kind of exception can be thrown here.
The main propose of this catch is to prevent the App from crashing in my opinion.